apps/frontend/src/app/5_pages/AavePage/AavePage.tsx

```diff
@@ -70,15 +70,25 @@ const AavePage: FC = () => {
     }
   }, [reserves, userReservesSummary]);
 
-  const lendPools: LendPoolDetails[] = useMemo(
-    () =>
-      reserves.map(r => ({
+  const lendPools: LendPoolDetails[] = useMemo(() => {
+    if (!userReservesSummary) {
+      return reserves.map(r => ({
         asset: r.symbol,
         apy: Decimal.from(r.supplyAPY).mul(100),
         canBeCollateral: r.usageAsCollateralEnabled,
-      })),
-    [reserves],
-  );
+        walletBalance: Decimal.from(0),
+      }));
+    } else {
+      return reserves.map(r => ({
+        asset: r.symbol,
+        apy: Decimal.from(r.supplyAPY).mul(100),
+        canBeCollateral: r.usageAsCollateralEnabled,
+        walletBalance:
+          userReservesSummary.balances.find(b => b.asset.symbol === r.symbol)
```

Comment:
A few things to considere here

- if reserves doesn't exist it will crash
- you can simplify the code doing just 1 map
- rect uses Object.is to compare dsps, memo will not detect any changes on the array at least it changes their memory location.
- normalize data at a higher architectural layer rather than within components to reduce complexity and dependencies on React.memo. By processing data normalization upstream, components can remain focused on rendering and be less concerned with data management intricacies. This practice not only improves the performance by reducing unnecessary re-renders but also enhances the maintainability of the code.

I'd do something like that (not tested just an example)
`
function normalizeBorrowPools(reserves: Reserve[], userReservesSummary?: UserReservesSummary): BorrowPoolDetails[] {
  return reserves.map(r => {
    const baseDetails = {
      asset: r.symbol,
      apy: Decimal.from(r.variableBorrowAPY).mul(100),
    };

    // Add additional details if userReservesSummary is available
    return userReservesSummary
      ? {
          ...baseDetails,
          available: userReservesSummary.borrowPower.div(r.priceInUSD),
          availableUSD: userReservesSummary.borrowPower,
        }
      : baseDetails;
  });
}

const borrowPools = normalizeBorrowPools(reserves, userReservesSummary);`

addiotional info
https://medium.com/suyeonme/react-lets-deep-dive-into-deps-array-of-useeffect-13ab96468db7

Reply:
I agree maybe it is missing a layer with some business logic since the operations require some validations and math calculations.

I saw some of them in utils which I personally found better than complex useMemos. However, depending on the case we could split even a bit more in another file (layer) to have them all together for each context.

Anyway, we can discuss it in more details the three of us if you want 💪

Reply:
Just pushed the normalizers @njaume suggested, I think looks pretty good, lemme know if you guys would do smth else here